packages/datadog-plugin-rhea/src/index.js

```diff
@@ -211,62 +89,27 @@ function getResourceNameFromMessage (msgObj) {
   return resourceName
 }
 
-function getResourceNameFromSender (sender) {
-  let resourceName = 'amq.topic'
-  if (sender.options && sender.options.target && sender.options.target.address) {
-    resourceName = sender.options.target.address
-  }
-  return resourceName
-}
-
 function getAnnotations (msgObj, tracer) {
   if (msgObj.message) {
-    return tracer.extract('text_map', msgObj.message.delivery_annotations)
+    const res = tracer.extract('text_map', msgObj.message.delivery_annotations)
+    return res
   }
 }
 
-function canTrace (link) {
-  return link.connection && link.session && link.session.outgoing
+function addDeliveryAnnotations (msg, tracer, span) {
+  if (msg) {
+    msg.delivery_annotations = msg.delivery_annotations || {}
+
+    tracer.inject(span, 'text_map', msg.delivery_annotations)
+  }
 }
 
-module.exports = [
-  {
-    name: 'rhea',
-    versions: ['>=1'],
-    file: 'lib/link.js',
-    patch ({ Sender, Receiver }, tracer, config) {
-      this.wrap(Sender.prototype, 'send', createWrapSend(tracer, config, this))
-      this.wrap(Receiver.prototype, 'dispatch', createWrapReceiverDispatch(tracer, config, this))
-    },
-    unpatch ({ Sender, Receiver }, tracer) {
-      this.unwrap(Sender.prototype, 'send')
-      this.unwrap(Receiver.prototype, 'dispatch')
-    }
-  },
-  {
-    name: 'rhea',
-    versions: ['>=1'],
-    file: 'lib/connection.js',
-    patch (Connection, tracer, config) {
-      this.wrap(Connection.prototype, 'dispatch', createWrapConnectionDispatch(tracer, config))
-    },
-    unpatch (Connection, tracer) {
-      this.unwrap(Connection.prototype, 'dispatch')
-    }
-  },
-  {
-    name: 'rhea',
-    versions: ['>=1'],
-    file: 'lib/session.js',
-    patch (Session, tracer, config) {
-      patchCircularBuffer(Session.prototype, this)
-    },
-    unpatch (Session, tracer) {
-      const CircularBuffer = Session[circularBufferConstructor]
-      if (CircularBuffer) {
-        patched.delete(CircularBuffer.prototype)
-        this.unwrap(CircularBuffer.prototype, 'pop_if')
-      }
-    }
+function getResourceNameFromSender (sender) {
+  let resourceName = 'amq.topic'
+  if (sender.options && sender.options.target && sender.options.target.address) {
```

Comment:
This should be normalized so that the subscriber can get the information without knowing the implementation details.